just_curious_game.ex

```diff
@@ -0,0 +1,325 @@
+    submitted_timeout
+    submitted_reminder
+    nominated_timeout
+    nominated_reminder
+    nominated_reattempts
+    answered_timeout
+    answered_reminder
+    answered_reattempts
+  )a
+
+  @moduledoc """
+  A `JustCurious.Game` is a FSM made up that transition through `round`s that to handle game state changes.
+
+  ## `transition`s
+  1. start
+  2. submitted
+  3. nominated
+  4. answered
+  5. finish
+
+  ## `round`s
+  1. submitting
+  2. nominating
+  3. answering
+  4. complete
+
+  ## `state`s
+  1. new    - start
+  2. active - submitted, nominated, answered
+  3. dead   - finish
+
+  ## Anatomy of the Game
+  Game state is tracked via a disposable Redis structure for preferences, participants, submissions, nominations, and answers
+
+  ``` plaintext
+  TXXX = Team, UXXX = User, 000.00X = Timestamp/Message UUID
+
+  team:T123:game:some-game
+    :players -> ["U111", "U222"]
+    :spectators -> ["U555"]
+    :invites -> ["U666"]
+    :joins -> ["U333"]
+    :name -> "Serene Desert"
+    :state -> "new"
+    :round -> "submitting"
+    :invitiation_messages -> [000.001] # Keep track of invitation messages for updating
+    :participant
+           :U111 # hash
+                :prompting -> "answer_confirmation" # prompt state tracking
+                :unconfirmed_submission -> "What's your favorite color?"
+                :submission -> "What's your favorite color?"
+                :answer -> "Green"
+                ...
+
+    :submissions # sorted set
+                :question-o-matic-1 -> 2 # a question-o-matic submitter with score
+                :U111               -> 1 # or real user submitter
+                :U222               -> 0
+
+    :nominatables # hash
+                 :000.001 -> U111 # nominatable message id and user id
+                 :000.002 -> U222
+    :nominations
+                :upvotes
+                          :U111
+                               :target -> "000.002" # user 1 upvoted second nominatable
+                               :reaction -> "heart" # with a heart reaction
+                          :U222
+                               :target -> "000.001"
+                               :reaction -> "+1" # with a heart reaction
+                :downvotes
+                          :U111 -> "000.003"
+
+  ```
+  """
+
+  @spec create(String.t, Map.t) :: %JustCurious.Game{}
+  def create(team, opts \\ %{}) do
+    name       = generate_name(team, :game)
+    game_id    = name |> String.downcase |> String.replace(" ", "-")
+    created_at = DateTime.utc_now |> DateTime.to_unix(:microseconds)
+
+    new_game = %JustCurious.Game{
+      team: team,
+      name: name,
+      channel: game_id,
+      round: opts[:round],
+      state: opts[:state] || :new,
+      created_at: created_at,
+      chainlink: 0
+    } |> Map.merge(JustCurious.GameConfigs.get(opts[:config]))
+
+    incr("team:#{team}:games_played")
+    hmset("team:#{team}:game:#{game_id}", redis_list_from_struct(new_game))
```

Comment:
These Redis commands with string interpolation everywhere seem a little inelegant but wrapping them all completely with customized functions may be more trouble than it's worth... I see below there are `get`ers and `set`ers that do something like this for commonly used commands, this seems good. Anyways, something to keep in mind.